packages/legend-graph/src/models/protocols/pure/v1/transformation/pureProtocol/serializationHelpers/V1_RuntimeSerializationHelper.ts

```diff
@@ -71,54 +73,81 @@ export const V1_legacyRuntimeModelSchema = createModelSchema(V1_LegacyRuntime, {
   ),
 });
 
-export const V1_identifiedConnectionModelSchema = createModelSchema(
-  V1_IdentifiedConnection,
-  {
+export const V1_identifiedConnectionModelSchema = (
+  plugins?: PureProtocolProcessorPlugin[],
+): ModelSchema<V1_IdentifiedConnection> =>
+  createModelSchema(V1_IdentifiedConnection, {
     connection: custom(
-      (val) => V1_serializeConnectionValue(val, true),
-      (val) => V1_deserializeConnectionValue(val, true),
+      (val) => {
+        if (plugins !== undefined) {
+          return V1_serializeConnectionValue(val, true, plugins);
+        }
+        return V1_serializeConnectionValue(val, true);
+      },
+      (val) => {
+        if (plugins !== undefined) {
+          return V1_deserializeConnectionValue(val, true, plugins);
+        }
+        return V1_deserializeConnectionValue(val, true);
+      },
     ),
     id: primitive(),
-  },
-);
+  });
 
-export const V1_storeConnectionModelSchema = createModelSchema(
-  V1_StoreConnections,
-  {
+export const V1_storeConnectionModelSchema = (
+  plugins?: PureProtocolProcessorPlugin[],
+): ModelSchema<V1_StoreConnections> =>
+  createModelSchema(V1_StoreConnections, {
     store: usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
     storeConnections: list(
-      usingModelSchema(V1_identifiedConnectionModelSchema),
+      usingModelSchema(
+        plugins !== undefined
+          ? V1_identifiedConnectionModelSchema(plugins)
+          : V1_identifiedConnectionModelSchema(),
+      ),
     ),
-  },
-);
+  });
 
-export const V1_engineRuntimeModelSchema = createModelSchema(V1_EngineRuntime, {
-  _type: usingConstantValueSchema(V1_RuntimeType.ENGINE_RUNTIME),
-  connections: list(usingModelSchema(V1_storeConnectionModelSchema)),
-  mappings: list(
-    usingModelSchema(V1_packageableElementPointerDeserrializerSchema),
-  ),
-});
+export const V1_engineRuntimeModelSchema = (
```

Comment:
Look at `V1_setupRelationalDatabaseConnectionModelSchema`, we should change this to `V1_setupEngineRuntimeModelSchema`, and call it when we setup the engine instead. Note that `createModelSchema`'s effect is global, which means that you don't need to pass this schema around, but rather you can just call `deserialize(V1_EngineRuntime, ...)` for example to make use of this.